modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/HorizontalCardTag.java

```diff
@@ -0,0 +1,383 @@
+		}
+
+		return _interactive;
+	}
+
+	public String getTitle() {
+		if ((_title == null) && (_horizontalCard != null)) {
+			return _horizontalCard.getTitle();
+		}
+
+		return _title;
+	}
+
+	public Boolean isDisabled() {
+		if (_disabled == null) {
+			if (_horizontalCard == null) {
+				return false;
+			}
+
+			return _horizontalCard.isDisabled();
+		}
+
+		return _disabled;
+	}
+
+	public Boolean isSelectable() {
+		if (_selectable == null) {
+			if (_horizontalCard == null) {
+				return false;
+			}
+
+			return _horizontalCard.isSelectable();
+		}
+
+		return _selectable;
+	}
+
+	public Boolean isSelected() {
+		if (_selected == null) {
+			if (_horizontalCard == null) {
+				return false;
+			}
+
+			return _horizontalCard.isSelected();
+		}
+
+		return _selected;
+	}
+
+	public void setActionDropdownItems(List<DropdownItem> actionDropdownItems) {
+		_actionDropdownItems = actionDropdownItems;
+	}
+
+	public void setData(Map<String, String> data) {
+		_data = data;
+	}
+
+	public void setDisabled(Boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setHorizontalCard(HorizontalCard horizontalCard) {
+		_horizontalCard = horizontalCard;
+	}
+
+	public void setHref(String href) {
+		_href = href;
+	}
+
+	public void setIcon(String icon) {
+		_icon = icon;
+	}
+
+	public void setInteractive(Boolean interactive) {
+		_interactive = interactive;
+	}
+
+	public void setSelectable(Boolean selectable) {
+		_selectable = selectable;
+	}
+
+	public void setSelected(Boolean selected) {
+		_selected = selected;
+	}
+
+	public void setTitle(String title) {
+		_title = title;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_actionDropdownItems = null;
+		_data = null;
+		_disabled = null;
+		_horizontalCard = null;
+		_href = null;
+		_icon = null;
+		_interactive = null;
+		_selectable = null;
+		_selected = null;
+		_title = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/HorizontalCard";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(
+		Map<String, Object> initialProps) {
+
+		Map<String, Object> props = super.prepareProps(initialProps);
+
+		props.put("actions", getActionDropdownItems());
+		props.put("cssClass", processCssClasses(new LinkedHashSet<>()));
+		props.put("disabled", isDisabled());
+		props.put("href", getHref());
+		props.put("icon", getIcon());
+		props.put("id", getId());
+		props.put("interactive", getInteractive());
+		props.put("selectable", isSelectable());
+		props.put("selected", isSelected());
+		props.put("title", getTitle());
+
+		return props;
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		if (isSelectable()) {
+			cssClasses.add("form-check");
+			cssClasses.add("form-check-card");
+			cssClasses.add("form-check-middle-left");
+		}
+		else {
+			cssClasses.add("card");
+			cssClasses.add("card-horizontal");
+		}
+
+		cssClasses.add("card-type-directory");
```

Comment:
I most cases, including this one, I followed the behavior of Clay 2 component, recreating it with Clay 3 tools. If not, we are potentially introducing backwards compatibility issues. In [Clay 2 component](https://github.com/liferay/clay/blob/2.x/packages/clay-card/src/ClayHorizontalCard.soy#L28), this class is always set.

Sidenote - sometimes breaking compatibility in my opinion is ok, in case that matching behavior is detrimental or very difficult to recreate. This is the case of dropdown markup and styles.

Reply:
I don't think that's the idea. We should provide backwards compatibility in terms of Tag API, nothing else, here we're using Clay v3. Mixing v2 and v3 markup just seems terrifying to me.

Reply:
I would argue that CSS classes and HTML markup **are** part of tag API. Who knows how many custom style modifications are there based on v2 classes and markup. Changing them is breaking backwards compatibility. Again, sometimes it is ok to break it, but that should be an exception.

@jbalsas could you weigh in on this?

Reply:
Since tags are really just there too render clay components and since the CSS and markup and components have changed I don't think we have lots of options here. 

>  Who knows how many custom style modifications are there based on v2 classes and markup.

I also think it wouldn't be the first time we change markup due to some library update, we can't really control that